RichText.cpp

```diff
@@ -74,6 +151,49 @@ void RichText::Line::draw(sf::RenderTarget &target, sf::RenderStates states) con
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+std::size_t RichText::Line::convertLinePosToLocal(std::size_t& pos) const
+{
+    // Trying to access something out-of-bounds is Undefined Behaviour.
+    // Let's not let that happen, OK?
+    assert(pos < getLength());
+    // Store our current position in the text vector
+    std::size_t arrayIndex = 0;
+    // Here, we select the right sf::Text to change
+    for (; pos >= m_texts[arrayIndex].getString().getSize(); ++arrayIndex)
+    {
+        pos -= m_texts[arrayIndex].getString().getSize();
+    }
+    return arrayIndex;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void RichText::Line::isolateCharacter(std::size_t pos)
+{
+    std::size_t localPos = pos;
+    std::size_t index = convertLinePosToLocal(localPos);
+    sf::Text copy = m_texts[index];
```

Comment:
Create a `sf::Text copy` which will act as the `temp` from below and just extract the string here, as all the calls below of `copy` start with `getString()` so we're actually interested in the string and not the `sf::Text` object.

Reply:
Now this is change is dangerous since I can't test that works... I really should install SFML and a C++ compiler in my computer again.